src/metabase/lib/order_by.cljc

```diff
@@ -0,0 +1,74 @@
+(ns metabase.lib.order-by
+  (:require
+   [clojure.spec.alpha :as s]
+   [metabase.lib.dispatch :as lib.dispatch]
+   [metabase.lib.field :as lib.field]
+   [metabase.lib.options :as lib.options]
+   [metabase.lib.schema :as lib.schema]
+   [metabase.lib.schema.order-by :as lib.schema.order-by]
+   [metabase.lib.util :as lib.util]
+   [metabase.util.malli :as mu]))
+
+(defmulti ^:private ->order-by-clause
+  {:arglists '([query stage-number x])}
+  (fn [_query _stage-number x]
+    (lib.dispatch/dispatch-value x)))
+
+(defmethod ->order-by-clause :asc
+  [_query _stage-number clause]
+  (lib.options/ensure-uuid clause))
+
+(defmethod ->order-by-clause :desc
+  [_query _stage-number clause]
+  (lib.options/ensure-uuid clause))
```

Comment:
Open to alternative syntax suggestions here. We're already going down the rabbit hole of making incompatible changes in pMBQL and we can always normalize them out before the QP sees them. 

Also worth mentioning that this PR isn't supposed to be a final design that we need to hammer absolutely everything out in. I kept `:asc` and `:desc` like this because that's how they work in MBQL, we can change that stuff as a follow on if we want